### PR TITLE
add test for more mutation operations

### PR DESCRIPTION
**Pull request by Toromtomtom**
This adds some tests for the remaining mutation operations. I'm not sure whether it makes sense to test all public methods, as some of them simply forward calls to each other.

### CODE REVIEW

**Comment by sageserpent-open** [comment]
Looks good to me.

Two observations:-

1. I would like to see the empty collection also used as a test case for the addition tests, though - given that the removal operations treat attempts to remove non-existent items as a no-operation, could this be added in without breaking the existing tests? The only exception would be `it should "be unchanged when removing elements that do not exist"`, which could be cutover to using `newElem` instead of using `b.head` and `b.removalAll(elem)` as a way of ensuring that the sut doesn't already contain the element being removed.

2. I'll put in an inline comment about the tests for updating.

File: src/test/scala/scala/collection/scalatest/mutable/MutableBagBehaviours.scala [hunk 1]
```diff
@@ -63,5 +63,59 @@ trait MutableBagBehaviours {
...
+      assert(b.multiplicity(newElem) == 3)
+    }
+
+    it should "support changing the multiplicity with setMultiplicity" in {
+      val b = bag
+      val elem = b.head
+      b.setMultiplicity(elem, 3)
+      assert(b.multiplicity(elem) == 3)
+      b.setMultiplicity(elem, 5)
+      assert(b.multiplicity(elem) == 5)
```
Comment (sageserpent-open) [inline src/test/scala/scala/collection/scalatest/mutable/MutableBagBehaviours.scala]: How about adding:-
`      b.setMultiplicity(elem, 0)`
`      assert(!b.contains(elem))`

here?

File: src/test/scala/scala/collection/scalatest/mutable/MutableBagBehaviours.scala [hunk 2]
```diff
@@ -63,5 +63,59 @@ trait MutableBagBehaviours {
...
+    }
+
+    it should "support changing the multiplicity with update" in {
+      val b = bag
+      val elem = b.head
+      b.update(elem, 3)
+      assert(b.multiplicity(elem) == 3)
+      b.update(elem, 5)
+      assert(b.multiplicity(elem) == 5)
```
Comment (sageserpent-open) [inline src/test/scala/scala/collection/scalatest/mutable/MutableBagBehaviours.scala]: Likewise:-
`      b.update(elem, 0)`
`      assert(!b.contains(elem))`

**Comment by sageserpent-open** [comment]
@Toromtomtom I guess it's on me now to deliver on the tests for pure-functional '-' tests. :smile:
Leave it with me...

**Comment by sageserpent-open** [comment]
Regarding my comment above about including the empty bag test case in the `MutableBagBehaviours` behaviour, I've realised that will still break the existing tests for element removal, simply because the assertions do expect an element to be removed, even if removal is a no-operation for the empty bag.

I guess a better way would be split the behaviours into those that would hold for all bags, and those that require the bag to be non-empty. So additive tests go in the first (as does `it should "be unchanged when removing elements that do not exist"`), whereas the subtractive ones go in the second.

**Comment by Toromtomtom** [comment]
I updated the code and separated the tests into two methods for empty and non-empty mutable bags.

**Comment by sageserpent-open** [comment]
@nicolasstucki If I can, I'll merge this next week unless you want to continue the review...

**Comment by nicolasstucki** [comment]
@sageserpent-open, you can merge it whenever you believe it is good

**Pull request merged**